src/naive/internals.rs

```diff
@@ -18,11 +18,27 @@
 use crate::Weekday;
 use core::fmt;
 
+#[cfg(feature = "rkyv")]
+use rkyv::{Archive, Deserialize, Serialize};
+
 /// The internal date representation: `year << 13 | Of`
-pub(super) type DateImpl = i32;
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[cfg_attr(feature = "rkyv", derive(Archive, Deserialize, Serialize))]
+#[allow(unreachable_pub)] // must be pub for rkyv
```

Comment:
`// must be pub for rkyv` seems surprising, why is this? There should really be a way for private types to be `Archive` etc.